Src/IronPython/Compiler/Ast/TryStatement.cs

```diff
@@ -114,6 +115,7 @@ public class TryStatement : Statement {
                 result =
                     Ast.Block(
                         Ast.Assign(runElse, AstUtils.Constant(true)),
+                        Ast.Assign(previousExceptionContext, Ast.Call(AstMethods.SaveCurrentException)),
```

Comment:
I left this in the try unlike in your implementation. I think that in either case we can't save the exception context like this because the underlying exception can change. It needs some sort of callback that can be made to restore the exception correctly if in a generator...